File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    count = 0
    print(number)
    if number <= 0: 
        raise ValueError('Only positive integers are allowed')
    elif number == 1 :
        return(0)
    while number != 1 :
        if number % 2 == 0 :
            number = number/2
            count+=1
            print(number,count)
        else : 
            number = (number*3)+1
            count += 1
            print(number,count)
    return(count)

print(steps(16))

File: c/collatz-conjecture/collatz_conjecture.c
#include <stdio.h>
#include "collatz_conjecture.h"

int steps (int start) {
    int count = 0 ;
    while (start != 1) {
        if (start % 2 == 0) {
            start = start / 2 ;
        }
        else {
            start = (start * 3) + 1 ;
        }
        count += 1 ;
    }
    return count ;
}

int main(void) {
    int result = steps(12);
    printf("steps(12) = %d\n", result);
    return 0;
}